src/pages/ParticipantsPage.js

```diff
@@ -0,0 +1,140 @@
+import OptionsList from '../components/OptionsList';
+import ROUTES from '../ROUTES';
+import CONST from '../CONST';
+
+const matchType = PropTypes.shape({
+    params: PropTypes.shape({
+        // reportID passed via route /participants/:reportID
+        reportID: PropTypes.string,
+    }),
+});
+
+const myPersonalDetailsType = PropTypes.shape({
+    // Display name of the current user from their personal details
+    displayName: PropTypes.string,
+
+    // Avatar URL of the current user from their personal details
+    avatar: PropTypes.string,
+
+    // login of the current user from their personal details
+    login: PropTypes.string,
+});
+
+const propTypes = {
+    /* Onyx Props */
+    // The personal details of the person who is logged in
+    myPersonalDetails: myPersonalDetailsType.isRequired,
+
+    // List of reports
+    reports: PropTypes.shape({
+        icons: PropTypes.arrayOf(PropTypes.string),
+        reportName: PropTypes.string,
+        participants: PropTypes.arrayOf(PropTypes.string),
+    }).isRequired,
+
+    // Route params
+    route: matchType.isRequired,
+
+    priorityMode: PropTypes.string.isRequired,
+};
+
+const getAllParticipants = (report, personalDetail) => {
+    const {icons, participants, reportName} = report;
+    const displayNames = reportName.split(', ');
+
+    const otherMembers = participants.map((login, index) => (
+        {
+            alternateText: login,
+            icons: [icons[index]],
+            keyForList: `${index}`,
+            tooltipText: login,
+            text: displayNames[index],
+            login,
+            displayName: displayNames[index],
+        }
+    ));
+
+    const final = [...otherMembers,
+        {
+            icons: [personalDetail.avatar],
+            text: personalDetail.displayName,
+            login: personalDetail.login,
+            alternateText: personalDetail.login,
+            keyForList: `${otherMembers.length}`,
+            tooltipText: personalDetail.login,
+            displayName: personalDetail.displayName,
+        },
+    ];
+
+    return final.map(item => ({...item, participantsList: [item]}));
+};
+
+const getSections = participants => ([{
+    title: '',
+    data: participants,
+    shouldShow: true,
+    indexOffset: 0,
+}]);
+
+const ParticipantsPage = ({
+    myPersonalDetails, route, reports, priorityMode,
+}) => {
+    const participants = getAllParticipants(reports[`report_${route.params.reportID}`], myPersonalDetails);
+
+    return (
+        <ScreenWrapper>
+            <HeaderWithCloseButton
+                title="Participants"
+                onCloseButtonPress={Navigation.dismissModal}
+            />
+            <View
+                pointerEvents="box-none"
+                style={[
+                    styles.detailsPageContainer,
+                ]}
+            >
+                { participants.length ? (
+
+                    <OptionsList
+                        sections={getSections(participants)}
```

Comment:
I don't think the getSections function is necessary.

Reply:
```
sections={[{
    title: '', data: participants, shouldShow: true, indexOffset: 0,
}]}
```

Are you suggesting I simply add this inline?

Reply:
yup, thx